src/vs/platform/quickinput/browser/quickInput.ts

```diff
@@ -227,4 +230,75 @@ export class QuickInputService extends Themable implements IQuickInputService {
 			})
 		};
 	}
+
+	formatPinnedItems(storageKey: string, quickPick: IQuickPick<IQuickPickItem>, storageService: IStorageService, callback?: () => Promise<void>): void {
```

Comment:
How about `reopenQuickPick` instead of `callback`? `void | Promise<any>` would probably be more appropriate so sync is allowed and since we don't care about the return?